Latest including RedSparr0w's pull

StudentBlake opened the pull request:
Closes #5

StudentBlake committed:
Added cache, status, live progress for loading files

Added cache, status, live progress for loading files

StudentBlake committed:
Change to different version from source

StudentBlake committed:
Minor tweaks

StudentBlake committed:
Blank file name fix if not in DB

StudentBlake committed:
Rev bump

StudentBlake committed:
Processing status now works as intended

StudentBlake committed:
Updated picture and changelog

StudentBlake committed:
Allow users to move columns

StudentBlake committed:
Rev bump

StudentBlake committed:
Rev bump

StudentBlake committed:
[RedSparr0w] Merge 'Custom rename schemes'

StudentBlake committed:
Changed ListView columns and added basic sorting

StudentBlake committed:
Update changelog

StudentBlake committed:
Updated picture

StudentBlake committed:
Merge branch 'master' into master

StudentBlake committed:
Revert "Merge branch 'master' into master"

This reverts commit 0c0487807b68002b97da8ae055096c6868f4168f, reversing
changes made to a4e97d3eb6cd1337c710734d9df66081ad35b681.

StudentBlake committed:
Fixed case in which game doesn't exist in db.xml and sorted by ID

StudentBlake committed:
Removed link to keys

RedSparr0w commented:
Could you add `Closes #5` to this PR's description, so that the issue is automatically closed once this gets merged.
Nice work on all these PR's btw.
👍

StudentBlake commented:
Trying to make it better slowly. Giba has been busy I think, but thanks for the custom naming scheme!

RedSparr0w commented:
Also noticed the `You must Refresh first!` message seems to appear constantly unless i have just refreshed, almost as soon as i click anything else the message appears again.
Would it be worth just calling the refresh function before showing the `Are you sure you want to rename ALL of your xci files automatically` message?

StudentBlake commented:
That's intentional for now. Basically I have to disable the buttons and ask to refresh every time the main window loses focus. The problem is the UpdateFilesList is using a backgroundworker, so it won't wait until the list finishes updating before renaming. Since the rename feature depends on the list matching the file locations on the disk, it needs to be run before renaming in the case that someone changed/moved a file after refreshing.

StudentBlake committed:
Don't stretch the game image

StudentBlake committed:
Minor tweak

StudentBlake committed:
Refactor some code

StudentBlake committed:
Minor changes

RedSparr0w committed:
Select the "Custom" rename checkbox on text input focus

RedSparr0w committed:
Add comments

RedSparr0w commented:
Would this work?:

on initial load/refresh save fileList with [`Directory.GetFiles()`](https://msdn.microsoft.com/en-us/library/07wt70x2\(v=vs.110\).aspx)
in the rename function have a check to see if fileList is still equal to [`Directory.GetFiles()`](https://msdn.microsoft.com/en-us/library/07wt70x2\(v=vs.110\).aspx), otherwise show message `Games directory has changed, please refresh and try again.`

As it shouldn't use too many resources so shouldn't need to be done by the service worker.

*Note: I'm pretty new to C#*

RedSparr0w committed:
remember last custom text rename format

StudentBlake commented:
I'm new too, so as you can tell the code is a mess.

StudentBlake committed:
Merge pull request #1 from RedSparr0w/_custom_rename

update [Custom Rename]

RedSparr0w committed:
check files are still the same before running batch rename/trim

RedSparr0w committed:
fixup

RedSparr0w committed:
Speed up batch renames

Takes < 1s total now (excluding refresh).

RedSparr0w committed:
speed up batch trim function

StudentBlake committed:
Merge pull request #2 from RedSparr0w/_custom_rename

Check files are the same before batch rename/trim

RedSparr0w committed:
Simplify TrimXCI function

StudentBlake committed:
Revert "Merge pull request #2 from RedSparr0w/_custom_rename"

This reverts commit 077060ef09ba15814c883fcc9a6ac07bebd7ddac, reversing
changes made to 40ebee8403fbb4cd914d2ee59ce851cf4c4f84e6.

StudentBlake committed:
Revert "Revert "Merge pull request #2 from RedSparr0w/_custom_rename""

This reverts commit 323d62018ef5854ab205823244eef69da948cdfc.

StudentBlake committed:
Merge pull request #3 from RedSparr0w/speedup

Speedup Batch Rename & Trim

StudentBlake committed:
Thanks :+1:

StudentBlake committed:
Merge pull request #4 from RedSparr0w/trim_xci

Simplify TrimXCI function

RedSparr0w committed:
Update

update

RedSparr0w committed:
Revert "speed up batch trim function"

This reverts commit 3b49d8e5ca4bace32aec8c0f856129dd9ead8bf0.

StudentBlake committed:
Merge pull request #5 from RedSparr0w/batch_trim

Revert batch trim speedup

StudentBlake committed:
Version bump

StudentBlake committed:
Removed unnecessary textbox

StudentBlake committed:
Update changelog/picture/instructions

StudentBlake committed:
Change to base64 for FilePath

StudentBlake committed:
Universal cache, simplified some code, updated TODO, NSWDB rev support

Cache is handled by the XCI PackageID. Luckily, PackageIDs are unique and quick to get. This makes the cache universal since it only stores the PackageID and TitleID. Thanks LucaFraga for hacbuild code!
All of these changes may make the program slightly slower because the file sizes are not cached anymore (as well as the overhead for PackageID).

StudentBlake committed:
Added changelog

StudentBlake committed:
Revert hash validation until fix for newer games (such as Sonic Mania) is figured out

StudentBlake committed:
Update

StudentBlake committed:
Minor UI tweak

StudentBlake committed:
Initial eShop support (reads titleid using NXTools)

StudentBlake committed:
Download eShop icons

StudentBlake committed:
Thanks for NXTools!

StudentBlake committed:
Update changelog

StudentBlake committed:
Better eShop metadata support

StudentBlake committed:
Minor version bump

StudentBlake committed:
Reading NSP is now as fast as XCI (workaround), moved tools and cache into their own folders

StudentBlake committed:
Minor rev bump

StudentBlake committed:
Awaiting nstool to support non-ANSI (hacky workaround)

Ultimately we want to use something native, but until then we rely on nstool

StudentBlake committed:
Removed NSP renaming (not reliable yet)

StudentBlake committed:
Minor text fixes and latest nstoolmod

StudentBlake committed:
Probably final commit, no major changes

StudentBlake committed:
Update README.md
